API/Parser/Parser.cs

```diff
@@ -67,40 +67,39 @@ public static class Parser
         private static readonly Regex SpecialTokenRegex = new Regex(@"SP\d+",
             MatchOptions, RegexTimeout);
 
+        private const string SeriesExpr = $@"(?<Series>{Balanced}?)";
+        private const string VolumeExpr = @"(?<Volume>\d+(\.\d)?(-\d+(\.\d)?)?)";
+        private const string ChapterExpr = @"(?<Chapter>\d+(\.\d)?(-\d+(\.\d)?)?)";
+        private const string BalancedParen = @"(?:[^()]|(?<open>\()|(?<-open>\)))*?(?(open)(?!))";
```

Comment:
Can you explain what open here means and why we use it?

Reply:
I'm not getting it. When I test it out, (?:[^()\[\]]|(?<open>\()|(?<-open>\))|(?<openbracket>\[)|(?<-openbracket>\])) is worse that just using . for Series group matching. I don't understand the regex nor the benefit of adding such complexity.